### PR TITLE
remove if/else statements from test, add destroy condition to fail(it…

### PR DESCRIPTION
**Pull request by Ian84Be**
…em) function

### CODE REVIEW

File: enhancing/enhancer.js [hunk 1]
```diff
@@ -38,6 +38,10 @@ function fail(item) {
   if (enhancement >= 15) durability -=10;
   if (enhancement > 16) enhancement -=1;
 
+  if (durability <= 0) {
+    return 'message: Item has been destroyed';
```
Comment (DaftBeowulf) [inline enhancing/enhancer.js:42]: aha, not a bad idea. I was more leaning in the direction of resetting any negative durability values back to zero (since having a negative value doesn't make sense), but sending a custom message to the user is also great

File: enhancing/enhancer.js [hunk 2]
```diff
@@ -71,11 +75,23 @@ function get(item) {
   if (!name || typeof(durability) !== 'number' || typeof(enhancement) !== 'number') {
     return 'error: object must include name, durability, and enhancement properties';
   }
-
-  if (enhancement > 0) name = `[+${enhancement}] ${name}`;
+  const regEx = /^\[\+\d+\]/;
```
Comment (DaftBeowulf) [inline enhancing/enhancer.js:78]: very nice! regex can be tricky but it's always good to at least have some of the basic matcher syntax familiarized

**Review by DaftBeowulf — COMMENTED** [review]
Good work accounting for the extra edge case I mentioned with regex nonetheless, and refactoring some of your test code too. I did like that you had a single global object being tested originally though, and now your test file is much longer because you have to redeclare a new object to be tested in each unit test. You can make it as explicit as you want since there's no harm in it, but something I think is at least cool to know _how_ to do is to reset such a variable before each test: if you have your global variable but just declare it is `let elvinBow={}` for instance, right underneath it you could use jest's beforeEach method like `beforeEach(()=> {elvinBow={prop1, prop2, etc})` and before each individual test runs, will run that beforeEach method and reset the properties of the bow object. then for tests where you want to see what happens with different enhancement values, for example, you still can pass in elvinBow but _spread_ it in and reassign the properties that need to be changed. ex: `expect(enhancer.succeed({...elvinBow, enhancement:20}).enhancement).toBe(20)` if originally the bow had an enhancement value of 15 and you already ran it to make sure it incremented when passed through `succeed` normally. a syntax you might prefer that would save you a decent chunk of time, but I can still tell you have a grasp of how the tests need to be run now :+1:
